### PR TITLE
share an album to another cozy user

### PR DESCRIPTION
**Pull request by enguerran**
Thanks to the new /sharings API, we are able to send a link to an existing cozy user.
Just enter his email address and his cozy's instance address and click on 'Send' button.

This feature is a cutting-edge technology and could be upgraded soon.

Made with :heart: by :goat::dash:

### CODE REVIEW

**Comment by enguerran** [comment]
@GoOz a special request for you. Could you give me an help on the style?

Here is what we want:


And here is what I did:


(The `add permission to write` toggle is coming next)

**Comment by enguerran** [comment]
[(5) [PHO] Partage d'un album à un autre Cozy](https://trello.com/c/Pt0BkELF/364-5-pho-partage-dun-album-%C3%A0-un-autre-cozy)

**Comment by enguerran** [comment]
@goldoraf and @y-lohse 
Note that I use the _gonna-change_ tabs components from cozy-ui. I need to update something in the https://github.com/cozy/cozy-ui/pull/144 as I get some weird behavior...

File: src/ducks/sharing/index.js [hunk 1]
```diff
@@ -38,6 +38,44 @@ export const createPermSet = (id, doctype) =>
...
   })
 
+export const share = async (id, doctype, email, url) => {
+  const target = await cozy.client.fetchJSON('POST', '/sharings/recipient', {
+    email,
+    url
+  })
+
+  const sharings = await cozy.client.fetchJSON('POST', '/sharings/', {
+    desc: 'Share Ilyn Sarkisian',
```
Comment (goldoraf) [inline src/ducks/sharing/index.js]: ?

**Comment by GoOz** [comment]
@enguerran I'll take care of it, don't worry. I finish something and I'm on it.

File: src/ducks/sharing/index.js (hunk 1 above)
Comment (enguerran) [inline src/ducks/sharing/index.js]: The `/sharings` route accepts a `desc` property to add what I think to be a description (maybe for the email body sent to the recipient). As I don't know what to add, and as it is related to share, I though [her name](https://en.wikipedia.org/wiki/Cher) could be a good description.

Comment (y-lohse) [inline src/ducks/sharing/index.js]: https://fr.wikipedia.org/wiki/Cher_(artiste) :')
But also yeah maybe put the name of the album there or something^^

File: src/ducks/sharing/index.js [hunk 2]
```diff
@@ -38,6 +38,44 @@ export const createPermSet = (id, doctype) =>
     }
   })
 
+export const share = async (id, doctype, email, url) => {
+  const target = await cozy.client.fetchJSON('POST', '/sharings/recipient', {
+    email,
+    url
+  })
+
```
Comment (y-lohse) [inline src/ducks/sharing/index.js]: Maybe not for this PR, but if I recall correctly the plan was to use a dummy implementation of `io.cozy.contacts` here.

File: src/ducks/sharing/index.js [hunk 3]
```diff
@@ -38,6 +38,44 @@ export const createPermSet = (id, doctype) =>
...
+  const sharings = await cozy.client.fetchJSON('POST', '/sharings/', {
+    desc: 'Share Ilyn Sarkisian',
+    permissions: {
+      album: {
+        description: 'album',
+        type: doctype,
+        values: [
+          id
+        ]
```
Comment (y-lohse) [inline src/ducks/sharing/index.js]: You've left out the [`verbs`](https://github.com/cozy/cozy-stack/blob/master/docs/sharing.md#request) here, is it on purpose?

File: src/ducks/sharing/index.js (hunk 1 above)
Comment (GoOz) [inline src/ducks/sharing/index.js]: HAHAHA 🤣

File: src/ducks/sharing/index.js (hunk 2 above)
Comment (enguerran) [inline src/ducks/sharing/index.js]: Not anticipating needs is the key of every successful development, said approximately Kent Beck.

File: src/ducks/sharing/index.js (hunk 3 above)
Comment (enguerran) [inline src/ducks/sharing/index.js]: I did not rely on the documentation which need to be updated as @Gara64 and julien said. So I cargo culted the https://github.com/Gara64/Sharotronic code.

File: src/ducks/sharing/ShareModal.jsx [hunk 4]
```diff
@@ -124,4 +112,111 @@ export class ShareModal extends Component {
...
+      <div><label for='' className={styles['coz-form-desc']}>{t('Albums.share.shareByLink.desc')}</label></div>
+      <div>
+        <Toggle
+          id='pho-album-share-toggle'
+          name='share'
+          checked={active}
+          onToggle={onToggle} />
+      </div>
+    </div>
+  </div >
```
Comment (enguerran) [inline src/ducks/sharing/ShareModal.jsx]: Why is it red?

File: src/ducks/sharing/index.js (hunk 1 above)
Comment (enguerran) [inline src/ducks/sharing/index.js]: @y-lohse you are right, but I cannot avoid such a good joke.

File: src/ducks/sharing/ShareModal.jsx (hunk 4 above)
Comment (y-lohse) [inline src/ducks/sharing/ShareModal.jsx]: Because of the white space after the tag name ;)

File: src/ducks/sharing/index.js (hunk 3 above)
Comment (Gara64) [inline src/ducks/sharing/index.js]: Leaving the `verbs` empty implies an ALL grant (POST,GET,PUT,DELETE). And FYI, I'm currently making the doc up-to-date :)

File: src/ducks/sharing/ShareModal.jsx (hunk 4 above)
Comment (enguerran) [inline src/ducks/sharing/ShareModal.jsx]: bien vu !

**Comment by enguerran** [comment]
@GoOz be careful, I rebased the commits, so you have to rebase too before adding your style's commit.

**Comment by GoOz** [comment]
@enguerran Roger that!

**Pull request merged**